src/engine/Service.cpp

```diff
@@ -504,3 +495,117 @@ std::optional<std::string> Service::idToValueForValuesClause(
       }
   }
 }
+
+// ____________________________________________________________________________
+void Service::precomputeSiblingResult(std::shared_ptr<Operation> left,
+                                      std::shared_ptr<Operation> right,
+                                      bool rightOnly, bool requestLaziness) {
+  AD_CORRECTNESS_CHECK(left && right);
+
+  auto skipSortOperation = [](std::shared_ptr<Operation>& op) {
+    auto children = op->getChildren();
+    if (children.size() == 1) {
+      op = children[0]->getRootOperation();
+    }
+  };
```

Comment:
That is definitely wrong. There are many other operations with only one child, in particular GROUP BY, DISTINCT, FILTER.... So you have to check whether it's a sort via a `dynamic  cast`.